waku/waku_keystore/keystore.nim

```diff
@@ -128,60 +128,19 @@ proc addMembershipCredentials*(path: string,
     if jsonKeystore.hasKey("credentials"):
 
       # We get all credentials in keystore
-      var keystoreCredentials = jsonKeystore["credentials"]
-      var found: bool
-
-      for membershipCredential in credentials:
-
-        # A flag to tell us if the keystore contains a credential associated to the input identity credential, i.e. membershipCredential
-        found = false
-
-        for keystoreCredential in keystoreCredentials.mitems():
-          # keystoreCredential is encrypted. We decrypt it
-          let decodedKeyfileRes = decodeKeyFileJson(keystoreCredential, password)
-          if decodedKeyfileRes.isOk():
-
-            # we parse the json decrypted keystoreCredential
-            let decodedCredentialRes = decode(decodedKeyfileRes.get())
-
-            if decodedCredentialRes.isOk():
-              let keyfileMembershipCredential = decodedCredentialRes.get()
-
-              # We check if the decrypted credential has its identityCredential field equal to the input credential
-              if keyfileMembershipCredential.identityCredential == membershipCredential.identityCredential:
-                # idCredential is present in keystore. We add the input credential membership group to the one contained in the decrypted keystore credential (we deduplicate groups using sets)
-                var allMemberships = toSeq(toHashSet(keyfileMembershipCredential.membershipGroups) + toHashSet(membershipCredential.membershipGroups))
-
-                # We sort membership groups, otherwise we will not have deterministic results in tests
-                allMemberships.sort(sortMembershipGroup)
-
-                # we define the updated credential with the updated membership sets
-                let updatedCredential = MembershipCredentials(identityCredential: keyfileMembershipCredential.identityCredential, membershipGroups: allMemberships)
-
-                # we re-encrypt creating a new keyfile
-                let encodedUpdatedCredential = updatedCredential.encode()
-                let updatedCredentialKeyfileRes = createKeyFileJson(encodedUpdatedCredential, password)
-                if updatedCredentialKeyfileRes.isErr():
-                  return err(KeystoreCreateKeyfileError)
-
-                # we update the original credential field in keystoreCredentials
-                keystoreCredential = updatedCredentialKeyfileRes.get()
+      let keystoreCredentials = jsonKeystore["credentials"]
+      let key = membership.hash()
+      if keystoreCredentials.hasKey(key):
+        # noop
+        return ok()
 
-                found = true
+      let encodedMembershipCredential = membership.encode()
+      let keyfileRes = createKeyFileJson(encodedMembershipCredential, password)
+      if keyfileRes.isErr():
+        return err(KeystoreCreateKeyfileError)
```

Comment:
This looks very nice. However, we are missing error details as per how the `KeystoreResult` is defined. I think is better to append the `keyfileRes.error` in this case. Something similar happens in the exception captured below, and elsewhere the `KeystoreResult` is used.

Reply:
Addressed in b4620326